src/node_watcher.js

```diff
@@ -270,16 +270,16 @@ NodeWatcher.prototype.processChange = function(dir, event, file) {
   var fullPath = path.join(dir, file);
   var relativePath = path.join(path.relative(this.root, dir), file);
   fs.lstat(fullPath, function(error, stat) {
+    var registered = this.registered(fullPath);
     if (error && error.code !== 'ENOENT') {
       this.emit('error', error);
     } else if (!error && stat.isDirectory()) {
       // win32 emits usless change events on dirs.
-      if (event !== 'change') {
+      if (event !== 'change' && registered) {
```

Comment:
Well, we can actually prune directories that couldn't possible match a given set of globs.